library/src/test/java/io/ona/kujaku/services/TrackingServiceTest.java

```diff
@@ -0,0 +1,290 @@
+                TrackingService.getIntent(context, MapActivity.class, new TrackingServiceHighAccuracyOptions()));
+
+        LocationManager locationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+        ShadowLocationManager shadowLocationManager = shadowOf(locationManager);
+        shadowLocationManager.setProviderEnabled(GPS_PROVIDER, true);
+
+        float distance = locationDeparture.distanceTo(location_1);
+        assertEquals(distance, 5, 1); // 5 meters +-1 meter
+
+        distance = location_1.distanceTo(location_2);
+        assertEquals(distance, 5, 1); // 5 meters +-1 meter
+
+        CountDownLatch latch1 = new CountDownLatch(1);
+        CountDownLatch latch2 = new CountDownLatch(1);
+
+        controller.get().registerTrackingServiceListener(new TrackingServiceListener() {
+            @Override
+            public void onFirstLocationReceived(Location location) {
+                assertEquals(location.getLatitude(), locationDeparture.getLatitude(),0);
+                assertEquals(location.getLongitude(), locationDeparture.getLongitude(),0);
+                latch1.countDown();
+            }
+
+            @Override
+            public void onNewLocationReceived(Location location) {
+                assertNotNull(location);
+                latch2.countDown();
+            }
+
+            @Override
+            public void onCloseToDepartureLocation(Location location) {
+                assertNotNull(location);
+            }
+
+
+            @Override
+            public void onServiceConnected(TrackingService service) {
+                // Empty body
+            }
+
+            @Override
+            public void onServiceDisconnected() {
+                // Empty body
+            }
+        });
+
+        controller.create().startCommand(0,0);
+
+        Thread.sleep(2000); // TaskService thread waiting for running
+
+        locationDeparture.setTime(System.currentTimeMillis());
+        shadowLocationManager.simulateLocation(locationDeparture);
+        latch1.await();
+
+        Thread.sleep(1000);
+        location_1.setTime(System.currentTimeMillis());
+        shadowLocationManager.simulateLocation(location_1);
+
+        Thread.sleep(1000);
+        location_2.setTime(System.currentTimeMillis());     // register location_1
+        shadowLocationManager.simulateLocation(location_2);
+        latch2.await();
+        Thread.sleep(1000);
+
+        location_3.setTime(System.currentTimeMillis());     // register location_2
+        shadowLocationManager.simulateLocation(location_3);
+        Thread.sleep(1000);
+
+        locationDeparture.setTime(System.currentTimeMillis()); // register location_3
+        shadowLocationManager.simulateLocation(locationDeparture);
+
+        List<Location> list = controller.get().getRecordedLocations();
+        assertEquals(list.size(), 4);
+
+        assertEquals(list.get(0).getLatitude(), location_1.getLatitude(),0);
```

Comment:
1. First location received by the GPS is locationDeparture. This point is not recorded but inform the listener that First point from GPS is received. You can begin walking.
2. We get a location in the window interval (location_1). It is registered as "pending location"
3. We get another location (location_2)  5 meters after location_1. We record location_1, location_2 is in our new interval window, it is registered as "pending location"
4. We get a new location (location_3) 5 meters after location_2. We record location_2, location_3 is in our new interval window, it is registered as "pending location"
5. we get now a point far from location_3, we register location3 and this new location as it is not in the window and we will not get a better point (better accuracy). 

Finally we have 4 locations registered (location_1, location_2, location_3, location_departure)

Reply:
Again, why don't we record the departure location as the first location? `location_1` which is pretty far away from location_1 (Approximately 5.6 metres away)

I assumed that the departure point is supposed to be the first point during tracking?

Reply:
The first locations we receive from the GPS are usually not accurate. You can try by using a new phone that never got Locations from GPS. It just indicates that GPS begins to give us Locations. We decide to not registered it . If the user want to take this first point, it can force it.

Reply:
Cool, I hope calling `kujakuMapView.trackingServiceTakeLocation()` will work inside `onFirstLocationReceived`.

Kindly add this to the documentation on the method and in the README since it's not so obvious and seems against the norm

Reply:
yes it works i just tested it.